search/mongo_search.go

```diff
@@ -14,6 +14,8 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
+var opInterruptedCode = 11601
```

Comment:
But to answer your other question, this is a know and well-documented approach to maintaining database integrity if long-running operations are in play:
https://blog.mlab.com/2014/02/mongodb-currentop-killop/
https://dzone.com/articles/finding-and-terminating-long